### ISSUE TITLE
About reproduction Results of SALAD + MeshVPR on train_set_hq

### ISSUE THREAD

**Issue by tony19980810** [gmberton/MeshVPR#4]
Hi,
Thank you for your meaningful work and for open-sourcing such high-quality code and datasets. I am very interested in your research. However, I encountered some issues while training SALAD + MeshVPR on train_set_hq, and here is my log:

2025-11-25 00:18:43 G:/meshvpr/MeshVPR/train.py
2025-11-25 00:18:43 Arguments: Namespace(method='salad', batch_size=32, num_epochs=5, iterations_per_epoch=10000, lr=1e-05, train_on_southern_half=False, infer_batch_size=16, resume_model=None, device='cuda', num_workers=8, num_preds_to_save=0, save_only_wrong_preds=False, log_dir=WindowsPath('logs/default/2025-11-25_00-18-43'), real_train_dir='F:\meshvpr/train_set_hq/real_database', synt_train_dir='F:\meshvpr/train_set_hq\synt_database', test_dir='I:\meshvpr/test_set/')
2025-11-25 00:18:43 The outputs are being saved in logs\default\2025-11-25_00-18-43
2025-11-25 00:21:08 All synthetic images: 9174240
2025-11-25 00:21:10 Found 2 test sets, namely ['synt_melbourne', 'val_set']
2025-11-25 07:09:16 Epoch 00 in 6:48:05, loss = 0.0001729127, < val_set - #q: 134; #db: 139104 >: R@1: 88.8, R@5: 89.6
2025-11-26 06:52:58 Epoch 01 in 6:53:25, loss = 0.0001731742, < val_set - #q: 134; #db: 139104 >: R@1: 87.3, R@5: 89.6
2025-11-30 07:29:36 Epoch 02 in 7:01:27, loss = 0.0001739871, < val_set - #q: 134; #db: 139104 >: R@1: 86.6, R@5: 89.6
2025-12-01 08:02:39 Epoch 00 in 7:39:28, loss = 0.0001754320, < val_set - #q: 134; #db: 139104 >: R@1: 84.3, R@5: 89.6

I found that the performance did not improve with training using the default parameters in your configuration, so I reduced the iterations_per_epoch to 1000 for each epoch:

2025-12-01 22:42:49 G:/meshvpr/MeshVPR/train.py
2025-12-01 22:42:49 Arguments: Namespace(method='salad', batch_size=32, num_epochs=10, iterations_per_epoch=1000, lr=1e-05, train_on_southern_half=False, infer_batch_size=64, resume_model=None, device='cuda', num_workers=8, num_preds_to_save=0, save_only_wrong_preds=False, log_dir=WindowsPath('logs/default/2025-12-01_22-42-49'), real_train_dir='F:\meshvpr/train_set_hq/real_database', synt_train_dir='F:\meshvpr/train_set_hq\synt_database', test_dir='I:\meshvpr/test_set/')
2025-12-01 22:42:49 The outputs are being saved in logs\default\2025-12-01_22-42-49
2025-12-01 22:43:55 All synthetic images: 9174240
2025-12-01 22:43:57 Found 2 test sets, namely ['synt_melbourne', 'val_set']
2025-12-02 00:02:39 Epoch 00 in 1:18:42, loss = 0.0001731706, < val_set - #q: 134; #db: 139104 >: R@1: 88.1, R@5: 90.3
2025-12-02 01:17:54 Epoch 01 in 1:15:12, loss = 0.0001730923, < val_set - #q: 134; #db: 139104 >: R@1: 88.1, R@5: 90.3
2025-12-02 02:34:23 Epoch 02 in 1:16:28, loss = 0.0001729624, < val_set - #q: 134; #db: 139104 >: R@1: 88.1, R@5: 90.3
2025-12-02 03:50:06 Epoch 03 in 1:15:42, loss = 0.0001728502, < val_set - #q: 134; #db: 139104 >: R@1: 88.1, R@5: 90.3
2025-12-02 05:09:02 Epoch 04 in 1:18:54, loss = 0.0001729182, < val_set - #q: 134; #db: 139104 >: R@1: 88.1, R@5: 90.3
2025-12-02 06:24:58 Epoch 05 in 1:15:55, loss = 0.0001725807, < val_set - #q: 134; #db: 139104 >: R@1: 88.8, R@5: 90.3
2025-12-02 07:40:02 Epoch 06 in 1:15:02, loss = 0.0001727038, < val_set - #q: 134; #db: 139104 >: R@1: 88.8, R@5: 89.6
2025-12-02 08:55:55 Epoch 07 in 1:15:51, loss = 0.0001728348, < val_set - #q: 134; #db: 139104 >: R@1: 88.8, R@5: 89.6
2025-12-02 10:11:17 Epoch 08 in 1:15:20, loss = 0.0001728600, < val_set - #q: 134; #db: 139104 >: R@1: 88.8, R@5: 89.6
2025-12-02 11:25:43 Epoch 09 in 1:14:25, loss = 0.0001726918, < val_set - #q: 134; #db: 139104 >: R@1: 88.8, R@5: 89.6
2025-12-02 11:25:44 Trained for 10 epochs, in total in 12:42:55
2025-12-02 14:38:38 < synt_melbourne - #q: 1249; #db: 394632 >: R@1: 59.4, R@5: 72.9, R@10: 78.9, R@20: 84.0, R@25: 85.9, R@50: 91.1, R@100: 95.0

In my results, it seems that SALAD + MeshVPR did not show significant performance improvement with training. I would like to ask for your opinion: Have I overlooked some training parameters?
Thank you again for your work and high-quality open source. I wish you smooth progress in your research.

**Comment by gmberton** [comment]
Hi, your final results on Melbourne look good.

It is possible that training converges very fast, and even a few hundreds/thousands of iterations are enough to align the two models.

Can you try to compute an evaluation at the beginning of the training, i.e. before any training iteration? I'm expecting it to give worse results than the final model, which would confirm the hypothesis above.

**Comment by tony19980810** [comment]
Thank you for your reply. I will try your suggestions later. Wishing you all the best.
